### PR TITLE
add failsafes for config data from constants.json file #36

### LINKED ISSUE #36: Bug: Add failsafe code when ACC_ENG_TRACKING is missing from the constants config
### Issue Type

None

### Summary

_No response_

### Priority

None

### Description

Add failsafe code when ACC_ENG_TRACKING is missing from the constants config so that pages where the config is not needed don't break if we remove the configuration entirely.

### Affected Device Classes

_No response_

### Affected Browsers/Devices and Versions

_No response_

### Steps to Reproduce

When ACC_ENG_TRACKING is missing the page will be broken.

### Expected Results

- [ ] The code doesn't break if the value is not present on the constants file
- [ ] On the method `extractObjectFromArray` add a failsafe code to ensure that `data` is iterable  
- [ ] On the method `formatStringToArray` add a failsafe code to ensure that `inputString` is a string (and not undefined)  
- [ ] On the file `/scripts/validate-elements.js` ensure the code doens't break when `TOOLS_CONFIGS` is not set on the constants file
- [ ] Remove all occurrences in all markets where the config is set but empty so the code doesn't break

### Actual Results

The code breaks when the configuration is not present on the constants file

### Attachment

_No response_

### PR DESCRIPTION
**Pull request by jonatan-lledo-netcentric**
Fix #36

Add extra layers of checks and move to the proper place some variables to avoid executing them always.

Now, if any of the constants values is missing, the function that need that value doesn't execute at all

Test URLs:
- Before: https://main--vg-macktrucks-com--volvogroup.aem.page/
- After: https://36-failsafe-for-constants-config--vg-macktrucks-com--volvogroup.aem.page/

Other markets:

Canada:

- Before: https://main--macktrucks-ca--volvogroup.aem.page/
- After: https://36-failsafe-for-constants-config--macktrucks-ca--volvogroup.aem.page/

Mexico:

- Before: https://main--macktrucks-com-mx--volvogroup.aem.page/
- After: https://36-failsafe-for-constants-config--macktrucks-com-mx--volvogroup.aem.page/

### CODE REVIEW

File: blocks/dealer-locator/dealer-locator.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ export default async function decorate(block) {
   // add the zip code to the input search, if it is present
   const zipCode = hasZipLocation ? escapeHTML(searchParams.get('l')) : null;
   const datasource = block.textContent.trim();
+  if (!GOOGLE_API_KEY) return;
```
Comment (alexiscoelho) [inline blocks/dealer-locator/dealer-locator.js:22]: Why this?

Comment (jonatan-lledo-netcentric) [reply blocks/dealer-locator/dealer-locator.js:22]: now I saw that is in the old DL, I'll remove it

File: blocks/magazine-subscribe/magazine-subscribe.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ export default async function decorate(block) {
 
   const iframeLink = createElement('a', {
     classes: 'iframe-link',
-    props: { href },
+    props: { href: href || '#' },
```
Comment (alexiscoelho) [inline blocks/magazine-subscribe/magazine-subscribe.js:23]: why not an empty string?

File: scripts/common.js [hunk 3]
```diff
@@ -335,6 +335,7 @@ async function getConstantValues() {
  * @returns {Object} An parsed object with those values and keys
  */
 export const extractObjectFromArray = (data) => {
+  if (!data || !Array.isArray(data)) return {};
```
Comment (alexiscoelho) [inline scripts/common.js:338]: Just `!Array.isArray(data)` should be enough

File: scripts/common.js [hunk 4]
```diff
@@ -457,6 +458,7 @@ export const getJsonFromUrl = async (route) => {
  *                     original input.
  */
 export const formatStringToArray = (inputString) => {
+  if (!inputString || typeof inputString !== 'string') return [];
```
Comment (alexiscoelho) [inline scripts/common.js:461]: Just type checking should be fine...

**Review by alexiscoelho — CHANGES_REQUESTED** [review]

File: blocks/dealer-locator/dealer-locator.js (hunk 1 above)
Comment (jonatan-lledo-netcentric) [reply blocks/dealer-locator/dealer-locator.js]: UPDATE: I add an empty string in case the api key is falsy

File: scripts/common.js (hunk 3 above)
Comment (jonatan-lledo-netcentric) [reply scripts/common.js]: done

File: scripts/common.js (hunk 4 above)
Comment (jonatan-lledo-netcentric) [reply scripts/common.js]: done

**Review by alexiscoelho — APPROVED** [review]

**Review by amglebihan — APPROVED** [review]

**Pull request merged**